engine/verification/assigner/engine.go

```diff
@@ -77,7 +77,7 @@ func (e *Engine) Done() <-chan struct{} {
 // is staked at the reference block of this execution receipt's result, chunk assignment is done on the execution result, and
 // the assigned chunks' locators are pushed to the chunks queue, which are made available for the chunk queue consumer (i.e., the
 // fetcher engine).
-func (e *Engine) handleExecutionReceipt(receipt *flow.ExecutionReceipt, containerBlockID flow.Identifier) {
+func (e *Engine) handleExecutionReceipt(ctx context.Context, receipt *flow.ExecutionReceipt, containerBlockID flow.Identifier) {
```

Comment:
Another way to deal with this if you don't want to pass context, is to get active trace from the tracer ( tracer keeps a list of active spans)

Reply:
see the one used for the consensus for details.

Reply:
If I understood the suggestion correctly, it requires to keep retrieving the same active span on the subsequent calls. In that case, I would still go with context, since it is easier to implement and maintain (e.g., no error handling is required of missing span). Also, we can use `WithSpanFromContext` (e.g., [here](https://github.com/onflow/flow-go/pull/466/files#diff-e7a28ce3f305916ae5d3992051dab0e249b4b7f155047057774b08199c9d37e8R267-R269)) to isolate the tracing logic from the business logic of the engine. This is primarily the same logic that we utilized in the current `finder` engine of VN, which this `assigner` engine is meant to replace.

Reply:
👍 then all good